website/preprints/tasks.py

```diff
@@ -62,3 +63,24 @@ def update_or_enqueue_on_preprint_updated(preprint_id, saved_fields=None):
             },
             celery=True
         )
+
+
+@celery_app.task(ignore_results=True, max_retries=5, default_retry_delay=CROSSREF_FAIL_RETRY_DELAY)
+def mint_doi_on_crossref_fail(preprint_id):
+    from osf.models import Preprint
+    preprint = Preprint.load(preprint_id)
+    existing_versions_without_minted_doi = Preprint.objects.filter(
+        versioned_guids__guid=preprint.versioned_guids.first().guid,
+        versioned_guids__version__lt=preprint.versioned_guids.first().version,
+        preprint_doi_created__isnull=True
+    ).exclude(id=preprint.id)
+    if existing_versions_without_minted_doi:
+        logger.error(
+            f'There are existing preprint versions for preprint with guid {preprint._id} that are missing DOIs. Versions: '
+            f'{list(existing_versions_without_minted_doi.values_list('versioned_guids__version', flat=True))}'
+        )
+        mint_doi_on_crossref_fail.retry(countdown=CROSSREF_FAIL_RETRY_DELAY)
```

Comment:
With `bind=True`, this should become
```suggestion
        self.retry()
```

Reply:
Also please add in `acks_late=True` to the decorator.

Reply:
Got this on merge